### PR TITLE
feat: redesign app

### PR DESCRIPTION
**Pull request by schmanu**
## Changes
- new theme
- remove safe-react-components
- update to MUI 5
- Use craco instead of react-rewired#
- Apply new Figma designs
- Remove tx history / reduce RPC calls a bit

## Todos

- [x] update dependencies
- [x] build EthHashInfo component
- [x] add balance column

### CODE REVIEW

File: src/components/ApprovalDialog.tsx [hunk 1]
```diff
@@ -76,34 +76,40 @@ export const ApprovalDialog = observer((props: ApprovalDialogProps) => {
...
-                          approval.setInputMode(id);
+                        value={approval.inputMode}
+                        onChange={(event) => {
+                          approval.setInputMode(event.target.value);
                         }}
-                      />
-                      <TextFieldInput
+                      >
+                        {changeApprovalItems.map((item) => (
+                          <MenuItem value={item.id}>{item.label}</MenuItem>
```
Comment (iamacook) [inline src/components/ApprovalDialog.tsx:100]:
```suggestion
                          <MenuItem key={item.id} value={item.id}>{item.label}</MenuItem>
```

File: src/components/approvallist/ApprovalEntry.tsx [hunk 2]
```diff
@@ -1,71 +1,79 @@
...
+        </div>
+      </FlexRowWrapper>
+      <FlexRowWrapper>
+        <Avatar
+          src={tokenMap?.get(approval.tokenAddress)?.logoUri}
+          sx={{ width: '24px', height: '24px' }}
+          alt={tokenMap?.get(approval.tokenAddress)?.symbol}
+        />
+        <div style={{ display: 'flex', flexDirection: 'column' }}>
+          <Typography fontWeight={700}>{tokenMap?.get(approval.tokenAddress)?.symbol}</Typography>
```
Comment (iamacook) [inline src/components/approvallist/ApprovalEntry.tsx:49]: Nit: if you move the `symbol` getter outside of the render, you'd need not call this twice.

File: src/components/approvallist/ApprovalHeader.tsx [hunk 3]
```diff
@@ -13,50 +14,65 @@ const StyledFirstRow = styled.div`
...
       </StyledFirstRow>
+      <Divider />
       <ColumnGrid
         style={{
-          paddingLeft: 22,
-          paddingRight: 44,
           minHeight: 48,
-          borderBottom: '2px solid #E8E7E6',
-          width: 'inherit',
+          borderBottom: `1px solid ${theme.palette.border.main}`,
```
Comment (iamacook) [inline src/components/approvallist/ApprovalHeader.tsx:53]: You can remove reliance on `useTheme` by using a callback here.

```suggestion
          borderBottom: (theme) => `1px solid ${theme.palette.border.main}`,
```

File: src/components/approvallist/ApprovalHeader.tsx [hunk 4]
```diff
@@ -13,50 +14,65 @@ const StyledFirstRow = styled.div`
...
-        <Title size="md">Approvals</Title>
-        <Settings />
+        <Typography variant="h4" fontWeight={700}>
+          Approvals
+        </Typography>
+        <Box display="flex" flexDirection="row" gap={2}>
+          <Settings />
+          <Button
+            variant={uiStore.selectedApprovals.length === 0 ? 'outlined' : 'contained'}
+            disabled={uiStore.selectedApprovals.length === 0}
```
Comment (iamacook) [inline src/components/approvallist/ApprovalHeader.tsx:41]: Nit: you could create a `hasSelectedApprovals` constant or similar to need not compare twice.

File: src/hooks/useDarkMode.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import { useEffect, useState } from 'react';
+
+const isSystemDarkMode = (): boolean => {
+  if (typeof window === 'undefined' || !window.matchMedia) return false;
```
Comment (iamacook) [inline src/hooks/useDarkMode.ts:4]: As you're not using NextJS, I don't think you need check for the existence of `window`. You might be able to simplify this implementation.

File: src/hooks/useDarkMode.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  return window.matchMedia('(prefers-color-scheme: dark)').matches;
+};
+
+export const useDarkMode = (): boolean => {
+  const [isDarkMode, setIsDarkMode] = useState<boolean>(false);
+
+  useEffect(() => {
+    const isDark = isSystemDarkMode();
+    setIsDarkMode(isDark);
+    document.documentElement.setAttribute('data-theme', isDark ? 'dark' : 'light');
```
Comment (iamacook) [inline src/hooks/useDarkMode.ts:14]: You aren't referencing this anywhere in your CSS at the moment. You can remove if if you want.

File: src/networks.ts [hunk 7]
```diff
@@ -14,17 +14,7 @@ export const networkInfo = new Map<number, NetworkInfo>([
...
-  ],
-  [
-    4,
-    {
-      chainID: 4,
-      name: 'Rinkeby',
-      shortName: 'rin',
-      currencySymbol: 'RIN',
-      baseAPI: 'https://safe-transaction.rinkeby.gnosis.io',
+      baseAPI: 'https://safe-transaction-mainnet.safe.global',
```
Comment (iamacook) [inline src/networks.ts:17]: Suggestion: fetching this from the config service.

File: src/stores/transactions/TransactionService.ts [hunk 8]
```diff
@@ -10,38 +10,35 @@ import { AccumulatedApproval } from './TransactionStore';
...
       })),
     );
 
-  return await Promise.all(
-    approvalLogs.map(async (log) => ({
-      ...log,
-      timestamp: (await web3Provider.getBlock(log.blockHash)).timestamp,
-    })),
-  );
+  console.log('Fetched approvalLogs');
```
Comment (iamacook) [inline src/stores/transactions/TransactionService.ts:40]: Not sure if you want to keep these logs?

File: src/networks.ts (hunk 7 above)
Comment (schmanu) [reply src/networks.ts:17]: I'd keep it out of scope for this PR.

**Pull request merged**